salesforce-composite/2.7/modules/ROOT/pages/edits/salesforce-composite-connector-examples.adoc

```diff
@@ -0,0 +1,469 @@
+		})
+	}),
+	"Contacts": {
+		"records": record.Contacts.records map ((record01, indexOfRecord01) -> {
+			"attributes": {
+				"type": record01.attributes.objectType,
+				"referenceId": record01.attributes.referenceId
+			},
+			"LastName": record01.LastName,
+			"Email": record01.Email,
+			"Title": record01.Title
+		})
+	}
+})
+]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+		<salesforce-composite:create-sobject-tree
+      doc:name="Create SObject tree"
+      config-ref="Salesforce_Composite_Salesforce_Composite"
+      type="Account"/>
+		<ee:transform doc:name="Transform Message">
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/json
+---
+payload]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+</mule>
+----
+
+== Pre-Query Example
+
+This example provides metadata for an object to query, generates two subrequests that contain this metadata, and then executes the subrequests:
+
+* HTTP *Listener*
++
+Accepts data from HTTP requests
+* *Transform Message*
++
+Specifies name for object
+* *Pre create*
++
+Provides metadata for the object that is going to be created and generates a subrequest used in execution
+* *Transform Message*
++
+Specifies values for fields
+* *Pre query*
++
+Provides metadata for the object that is going to be queried and generates a subrequest used in execution
+* *Transform Message*
++
+Passes output from *Pre query* operation
+* *Execute composite batch*
++
+Sends all subrequests in one batch
+* *Transform Message*
++
+Outputs the results in JSON format
++
+.The flow for this example includes an HTTP Listener component, Transform components, and multiple Salesfore Composite operations.
+image::salesforce-composite-prequery-flow.png[Pre-Query flow in Studio]
+
+=== Add an HTTP Listener
+
+The HTTP *Listener* source operation initiates the flow when it detects an event in the `executeFlow` path:
+
+. Create a new Mule project in Studio.
+. In the *Mule Palette* view, search for *HTTP* and select the *Listener* operation.
+. Drag the *Listener* operation onto the canvas.
+. In the Listener properties window, set the *Path* field to `/executeFlow`.
+
+=== Add the First Transform Message Component
+
+This *Transform Message* component specifies a name for the created object and passes the name to the *Pre create* operation:
+
+. In the *Mule Palette* view, search for *Transform Message*.
+. Drag the *Transform Message* component onto the canvas, to the right of *Listener*.
+. Enter the name `MyNewAccount`:
++
+[source,dataweave,linenums]
+----
+%dw 2.0
+%output application/java
+---
+{
+   Name: "MyNewAccount"
+}
+----
+
+=== Add the Pre Create Operation
+
+The *Pre create* operation provides metadata for creating an object called `NewAccount` and generates a subrequest for the `executeCompositeBatch` operation:
+
+. In the *Mule Palette* view, search for *Salesforce* and select the Salesforce Composite *Pre create* operation.
```

Comment:
GLOBALLY fix the "search for" strings to lowercase mono; I'll stop marking.

Reply:
Also GLOBALLY fix the names of the operations (plain init. cap.) vs. the UI labels for the UI elements (bold). I'll stop marking.